demo/app.js

```diff
@@ -226,11 +233,21 @@ class App extends React.Component {
               <input
                 id={ 'textIsEditableCheckbox' }
                 type="checkbox"
-                defaultChecked="true"
+                checked={ this.state.isTextEditable }
```

Comment:
A controlled component seems like the way to go here.

https://reactjs.org/docs/forms.html#handling-multiple-inputs
https://reactjs.org/docs/uncontrolled-components.html#default-values